pkg/controller/deployment.go

```diff
@@ -466,14 +466,6 @@ func (dc *controller) reconcileClusterMachineDeployment(key string) error {
 		return nil
 	}
 
-	if deployment.DeletionTimestamp == nil {
```

Comment:
I was thinking of keeping management of machineSets & machineDeployments controllers independent of underlying machineClass/secrets. There are several reasons why this would be helpful.

1. Theses controller won't bother about the `kind` of machine classes and their validation and their secrets fields. This helps us as these validations for machine classes may not be available at this controller and are part of machine controller. 

2. This makes sense to me as IMO the higher level controllers don't need to bother about these reference objects (machineClasses & secrets). That was we leave validation of such objects to lower level controllers. Also the machine controller may also be implemented with having the providerSpec inline without the use of reference objects if required (the way community does/used to do). 

3. We have observed operations (deletions) being blocked earlier due to such validations and this would prevent such scenarios.  

Hence have removed this here. WDYT?